addon/v1alpha1/types_addondeploymentconfig.go

```diff
@@ -58,6 +58,31 @@ type AddOnDeploymentConfigSpec struct {
 	// +kubebuilder:validation:MaxLength=63
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	AgentInstallNamespace string `json:"agentInstallNamespace,omitempty"`
+
+	// Resources required by add-on agents.
+	// If a container matches multiple ContainerResourceRequirements, the last one in the array will
+	// take precedence.
+	// +optional
+	// +listType=map
+	// +listMapKey=containerID
+	AgentResources []ContainerResourceRequirements `json:"agentResources,omitempty"`
+}
+
+// ContainerResourceRequirements defines resources required by one or a group of containers.
+type ContainerResourceRequirements struct {
+	// ContainerID is a unique identifier for an agent container. It consists of three parts: resource kind,
+	// resource name, and container name, separated by ':'. The format follows
+	// '{resource_kind}:{resource_name}:{container_name}'. Wildcards (*) can be used in any part to match
+	// multiple containers. For example, '*:*:*' matches all containers of the agent.
+	// +required
+	// +kubebuilder:validation:Required
+	// "+kubebuilder:validation:Pattern=`^.+:.+:.+$`
```

Comment:
does this work with the begining of `"`?

Reply:
The validation here is not strict. A value, like `":abc:def` is allowed.